src/server/json_family.cc

```diff
@@ -1485,7 +1485,12 @@ OpStatus OpMerge(const OpArgs& op_args, string_view key, std::string_view json_s
     op_args.shard->search_indices()->RemoveDoc(key, op_args.db_cntx, it_res->it->second);
 
     JsonType* obj = it_res->it->second.GetJson();
-    RecursiveMerge(*parsed_json, obj);
+    try {
+      RecursiveMerge(*parsed_json, obj);
+    } catch (const std::exception& e) {
+      LOG(WARNING) << "Exception in OpMerge: " << e.what();
+      return OpStatus::KEY_NOTFOUND;
```

Comment:
@romange not particularly happy with this. Any recommendations for actual OpStatus error ? Imo `KEY_NOTFOUND` is not good enough

Reply:
INVALID_VALUE?

 and please make it `LOG(ERROR)` and also provide more information like json values, so we could reproduce if this log ever happens.

Reply:
LOG_EVERY_T(ERROR,1)